### ISSUE TITLE
View a recipe in another unit system

### ISSUE THREAD

**Issue by JackMiranda** [JackMiranda/herms#28]
Howdy folks! Yeah, I actually say "howdy folks" in real life because I'm from the western United States. Other silly things we say out here include "add 3/4 cup of water" and "mix in 1/8 teaspoons cinnamon." This is no good if your measuring cups are in milliliters! 

Add a flag to specify viewing a recipe in a different unit system. Make a new file in `src` called `UnitConversions.hs` or something clever like that that contains functions to convert quantities in one unit to another. Start out with just the following units:

- 1 tsp   <=> 5 mL
- 1 Tbsp <=> 15 mL
- 1 cup  <=> 250 mL
- 1 oz    <=> 28 g

We can add on more from there, but that should cover the basics for now. 

Tips:

- Familiarize yourself with the `Recipe` data type
- Take a look at the `showRecipe` function
- Study how flags are handled

**Comment by nasanos** [comment]
I'll get started on this, if no one else has taken it, starting with the functions involved.

After looking over `Types.hs`, my impression is that `UnitConversions.hs` could maybe consist of two functions:
* A function converting metric units to imperial,
* A function converting imperial units to metric.

This would also result in all units, after the conversation, being of the same system.

Alternatively, this could be accomplished on a case-to-case basis, with a function for each unit:
* A function for tsp,
* A function for Tbsp,
* A function for mL,
* Etc.

This approach might be a little trickier, but it would give more control over the conversion implementation.

**Comment by nasanos** [comment]
My current pull request (34) doesn't solve all of the issues proposed here, but I think it gets a start in place, which I plan to continue on as time allows.

I was wondering if you could give me a rundown of how Herms uses flags. Hopping through the files still left me unsure of best-practices for implementing flags for the unit conversions.

**Comment by JackKiefer** [comment]
Great work! Herm's uses a library called optparse-applicative to handle all of the flags. I'm not actually super familiar with it; @vrom911 is our resident expert there :)

**Comment by vrom911** [comment]
@nasanos nice work so far! :1st_place_medal: 
Flags handling is actually very easy now with usage of [`optparse-applicative`](https://hackage.haskell.org/package/optparse-applicative) :relaxed: 
All code related to it is in `Main` module [here](https://github.com/JackKiefer/herms/blob/75e533ffa180f8295ad13f1d7590f6fee9f8b909/app/herms.hs#L274).
But first of all I suggest you to replace check of `1` or `0` to proper data type. Something like this:
```
data Conversion = Metric | Imperial
```
This will make code safer and easier to read and understand. But not only, this also will help to make adding flag process much easier and painless. 
But first let me clarify how we want the flag to be?
I vote for having default conversion as `Imperial` and change it to `Metric` only when flag `-m|--metric` (or any more convincing name) is present. If this is the case all you need is to write `conversionP :: Parser Conversion` which will be the [flag](https://hackage.haskell.org/package/optparse-applicative-0.14.0.0/docs/Options-Applicative.html#v:flag). 
Probably it's not the best choice, your suggestions are more than welcome. It's usually the hardest part to create the architecture of how it should be..
Anyway, if you have any questions I'm always happy to answer :slightly_smiling_face:

**Comment by JackKiefer** [comment]
Thanks @vrom911 ! 

Some good ideas, but I think that perhaps a wiser option would to have something like this:

```
data Conversion = Metric | Imperial | None
```

By default recipes could simply be labeled `None` and not attempt to convert themselves. Only when passed a flag like `-i|--imperial` or `-m|--metric` would the recipe be converted and shown. That way, users with nothing but metric recipes don't have to type `--metric` every time they want to view a recipe. :)

**Comment by nasanos** [comment]
So, it looks like this would take two flags, one for metric and one for imperial. I didn't see a way in the `optparse-applicative` docs to handle this otherwise, but correct me if I'm missing something there.

**Comment by vrom911** [comment]
@JackKiefer @nasanos  if we have now more than two cases it is not going to be switch anymore. I guess in this case in will be more comfortable for usage purposes to have something like `[--system SYSTEM_TYPE]`. And when you need to turn on any just type `herms view --system metric` for example. 
Also the good point of this choice is that in the future you would be able to add more metric systems without rebuilding the architecture and making number of options really crazy :)

**Comment by ppartarr** [comment]
I believe this should be closed since the `-convert` flag was merged in #38

**Comment by langston-barrett** [comment]
I agree. Let's follow up with other issues if there are still remaining tasks.
